### ISSUE TITLE
Failed to set advertising parameters

### ISSUE THREAD

**Issue by littleboot** [MasterDevX/lampify#3]
When I try to setup (or other commands) I get an error message.
```
$ ./lampify v setup
[I] Turning the lamp on
[E] Failed to set advertising parameters!
```

Looking at the code it fails when the `hciSetParams` function returns. I assume this might be caused by my Bluetooth device driver not supporting this function. not sure how to fix this, help would be appreciated.

**Comment by Guenda85** [comment]
same for me:
~/lampify$ ./lampify v setup
[I] Connecting to the lamp
[E] Failed to set advertising parameters!

Edit:
tried 2 OS with same result:
Linux gke-sv-arch 6.5.5-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 23 Sep 2023 22:55:13 +0000 x86_64 GNU/Linux
Linux GKE-SV-Linux 6.2.0-33-generic #33~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Sep  7 10:33:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

**Comment by yvesdm3000** [comment]
I had the same problem. Try to run it as root, it should fix your problem.

**Comment by Guenda85** [comment]
running as root comes up with different „error“:

~/lampify$ sudo ./lampify v setup
[W] Failed to send notification!
[I] Connecting to the lamp

**Comment by yvesdm3000** [comment]
That error is because it can't connect to libnotify, a system used by the Gnome desktop to show notifications as popups.

Do you have Gnome running as desktop environment? If not, better use "sudo ./lampify q setup"

**Comment by Guenda85** [comment]
yeah, just did it within terminal over ssh :-) executing it in quite mode leads me to:
sudo ./lampify q setup
[I] Connecting to the lamp

**Comment by Guenda85** [comment]
Unfortunately still no progress - no error so far but setup does not find/connect the lamp :-(

**Comment by mak448a** [comment]
I ran as root as well. The output displays nothing. Any updates?

**Comment by erfanium** [comment]
check my fork: https://github.com/erfanium/lampify

I fixed this issue
